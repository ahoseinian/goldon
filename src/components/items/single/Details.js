import React from 'react';

const Details = ({item}) => (
  <div>{JSON.stringify(item)}</div>
);

Details.propTypes = {
  item: React.PropTypes.object.isRequired
};

export default Details;
