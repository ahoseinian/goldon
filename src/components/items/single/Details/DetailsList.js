import React from 'react';
import SingleRowTable from '../../../common/SingleRowTable'
import LinkList from '../../../common/Links/LinkList'

const DetailsList = ({item}) => (
  <div>
    <SingleRowTable name="مدل">{item.modelsNames.join(' , ')}</SingleRowTable>
    <SingleRowTable name="رنگهای موجود">{item.colors.join(' , ')}</SingleRowTable>
    <SingleRowTable name="نام های دیگر">{item.names.join(' , ')}</SingleRowTable>
    <SingleRowTable name="لینک ها">
      {item.links.length && <LinkList links={item.links}/>}
    </SingleRowTable>
  </div>
);

DetailsList.propTypes = {
  item: React.PropTypes.object
};

export default DetailsList;
