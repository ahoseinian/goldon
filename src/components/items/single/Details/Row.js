import React from 'react';

const SingleRow = ({name, value, measure}) => (
  <tr>
    <th>{name}</th>
    <td>{value} {measure}</td>
  </tr>
);

SingleRow.propTypes = {
  name: React.PropTypes.string.isRequired,
  value: React.PropTypes.oneOfType([
    React.PropTypes.number,
    React.PropTypes.string
  ]),
  measure: React.PropTypes.string
};

export default SingleRow;
