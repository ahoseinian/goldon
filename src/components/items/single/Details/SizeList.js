import React from 'react';
import Row from './Row'

const SizeList = ({tool, arz, ertefa, ghotr}) => (
  <table className="table table-bordered mt-4">
    <tbody>
      <Row name="طول" value={tool} measure="سانتی متر"/>
      <Row name="عرض" value={arz} measure="سانتی متر"/>
      <Row name="ارتفاع" value={ertefa} measure="سانتی متر"/>
      <Row name="قطر" value={ghotr} measure="سانتی متر"/>
    </tbody>
  </table>
);

SizeList.propTypes = {
  tool: React.PropTypes.number,
  arz: React.PropTypes.number,
  ertefa: React.PropTypes.number,
  ghotr: React.PropTypes.number
};

export default SizeList;
