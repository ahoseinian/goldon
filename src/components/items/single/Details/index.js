import React from 'react';
import SimpleCard from '../../../common/card/SimpleCard'
import DetailsList from './DetailsList'
import SizeList from './SizeList'

const Details = ({item}) => (
  <div>
    <SizeList {...item.size}/>
    <DetailsList item={item}/>
    <SimpleCard title="نگهداری">{item.maintenance}</SimpleCard>
    <SimpleCard title="توضیحات">{item.information}</SimpleCard>
    <SimpleCard title="خواص">{item.benefits}</SimpleCard>
  </div>
);

Details.propTypes = {
  item: React.PropTypes.object.isRequired
};

export default Details;
