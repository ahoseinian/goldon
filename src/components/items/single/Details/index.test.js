import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import Details from './index'

const item = {
  size: {},
  modelsNames: [],
  colors: [],
  names: [],
}

describe('item > single> Details > index', () => {
  it('renders', () => {
    const wrapper = shallow(<Details item={item}/>)
    expect(toJson(
      wrapper
    )).toMatchSnapshot()
  })
})
