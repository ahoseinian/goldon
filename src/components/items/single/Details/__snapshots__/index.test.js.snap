exports[`item > single> Details > index renders 1`] = `
<div>
  <styled.div
    className="card-block">
    <SizeList />
    <DetailsList
      colors={Array []}
      modelsNames={Array []}
      names={Array []}
      size={Object {}} />
  </styled.div>
  <SimpleCard
    title="نگهداری" />
  <SimpleCard
    title="توضیحات" />
  <SimpleCard
    title="خواص" />
</div>
`;
